docs/source/reference/templates.md

```diff
@@ -1,28 +1,29 @@
 # Working with templates and UI
 
 The pages of the JupyterHub application are generated from
-[Jinja](http://jinja.pocoo.org/) templates. These allow the header, for
+[Jinja](https://jinja.palletsprojects.com/en/3.1.x/) templates. These allow the header, for
 example, to be defined once and incorporated into all pages. By providing
-your own templates, you can have complete control over JupyterHub's
+your own template(s), you can have complete control over JupyterHub's
 appearance.
 
 ## Custom Templates
 
-JupyterHub will look for custom templates in all of the paths in the
-`JupyterHub.template_paths` configuration option, falling back on the
+JupyterHub will look for custom templates in all paths included in the
+`JupyterHub.template_paths` configuration option, falling back on these
 [default templates](https://github.com/jupyterhub/jupyterhub/tree/HEAD/share/jupyterhub/templates)
-if no custom template with that name is found. This fallback
+if no custom template(s) with specified name(s) are found. This fallback
 behavior is new in version 0.9; previous versions searched only those paths
 explicitly included in `template_paths`. You may override as many
 or as few templates as you desire.
 
 ## Extending Templates
 
-Jinja provides a mechanism to [extend templates](http://jinja.pocoo.org/docs/2.10/templates/#template-inheritance).
-A base template can define a `block`, and child templates can replace or
-supplement the material in the block. The
-[JupyterHub templates](https://github.com/jupyterhub/jupyterhub/tree/HEAD/share/jupyterhub/templates)
-make extensive use of blocks, which allows you to customize parts of the
+Jinja provides a mechanism to [extend templates](https://jinja.palletsprojects.com/en/3.0.x/templates/#template-inheritance).
+
+A base template can define `block`(s) within itself that child(ren) templates can fill up or
```

Comment:
even plural, I'd call them 'child templates' rather than 'children templates':

```suggestion
A base template can define `block`(s) within itself that child templates can fill up or
```